lib/mvi/messages/find_candidate_message.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+require 'ox'
+require_relative 'message_builder'
+
+module MVI
+  module Messages
+    # Builds an MVI SOAP XML message.
+    #
+    # = Usage
+    # Call the .build passing in the candidate's first and last name, dob, and ssn.
+    #
+    # Example:
+    #  dob = Time.new(1980, 1, 1).utc
+    #  message = MVI::Messages::FindCandidateMessage.build('John', 'Smith', dob, '555-44-3333')
+    #
+    class FindCandidateMessage
+      extend MVI::Messages::MessageBuilder
+      EXTENSION = 'PRPA_IN201305UV02'
+
+      class << self
+        def build(first_name, last_name, dob, ssn)
+          validate(dob, first_name, last_name, ssn)
+          header(EXTENSION)
+          find_candidate_body(parameter_list(first_name, last_name, dob, ssn))
+          doc << envelope_body(@message)
+          Ox.dump(doc)
+        rescue => e
+          Rails.logger.error "failed to build find candidate message: #{e.message}"
+          raise
+        end
+
+        def validate(dob, first_name, last_name, ssn)
+          raise ArgumentError, 'names should be Strings' unless [first_name, last_name].all? { |i| i.is_a? String }
+          raise ArgumentError, 'dob should be a Time object' unless dob.is_a? Time
+          raise ArgumentError, 'ssn should be of format \d{3}-\d{2}-\d{4}' unless ssn =~ /\d{3}-\d{2}-\d{4}/
+        end
+
+        def find_candidate_body(parameter_list)
```

Comment:
prefixing vars with `internal` or finding a way to avoid the local variable assignment might be nice so those names aren't shadowed.